c7n/resources/rds.py

```diff
@@ -1568,3 +1568,49 @@ def process(self, resources, event=None):
                     results.append(resource)
                     break
         return results
+
+
+@actions.register('modify-db')
+class ModifyDb(BaseAction):
+    """Modifies an RDS instance based on specified param and value
+    using ModifyDbInstance.
+
+    'Parameter' determines which field is being modified.
+    'Value' is what the modified field will be set to.
+    'Immediate" determines whether the modification is applied immediately
+    or not. If 'immediate' is not specified, default is false.
+
+    :example:
+
+    .. code-block:: yaml
+
+            policies:
+              - name: disable-rds-deletion-protection
+                resource: rds
+                filters:
+                  - DeletionProtection: true
+                actions:
+                  - type: modify-db
+                    parameter: DeletionProtection
+                    value: false
+                    immediate: true
+    """
+
+    schema = type_schema(
+        'modify-db',
+        parameter={},
```

Comment:
type string at least, alternatively enumerate the potential values here.

Reply:
the other thought is setting multiple parameters in a single api call instead of having to do separate ones per parameter per resource.

Reply:
Modified parameter to type object to allow multiple parameters, didn't specify more as more complicated parameters such as CloudwatchLogsExportConfiguration and ProcessorFeatures require lists/dicts. Added some exception handling and also made parameter required.